core/src/main/scala/cats/data/EitherT.scala

```diff
@@ -335,6 +337,50 @@ object EitherT extends EitherTInstances {
    */
   final def liftT[F[_], A, B](fb: F[B])(implicit F: Functor[F]): EitherT[F, A, B] = right(fb)
 
+  /** Create an `EitherT` from a recoverable F[_].
+    *
+    * {{{
+    * scala> import cats.implicits._
+    * scala> val o1: Option[Int] = Some(5)
+    * scala> EitherT.fromRecoverable[Option, Unit](o1) { case _ => 1 }
+    * res0: EitherT[Option, Nothing, Int] = EitherT(Some(Right(5)))
+    * scala> val o2: Option[Int] = None
+    * scala> EitherT.fromRecoverable[Option, Unit](o2) { case _ => 1 }
+    * res1: EitherT[Option, Nothing, Int] = EitherT(Some(Right(1)))
+    * }}}
+    */
+  final def fromRecoverable[F[_], E]: FromRecoverablePartiallyApplied[F, E] = new FromRecoverablePartiallyApplied
+
+  /**
+    * Uses the [[http://typelevel.org/cats/guidelines.html#partially-applied-type-params Partially Applied Type Params technique]] for ergonomics.
+    */
+  private[data] final class FromRecoverablePartiallyApplied[F[_], E](val dummy: Boolean = true) extends AnyVal {
+    def apply[A, B](f: F[B])(pf: PartialFunction[E, B])(implicit me: MonadError[F, E]): EitherT[F, A, B] =
+      EitherT.right[A](me.recover(f)(pf))
+  }
+
+  /** Creates an `EitherT` from a Future, recovers its error into a Either.Left.
```

Comment:
Can we add a comment saying this will result in a Future that always succeeds but contains the potential error (and value) in the success. This has been a source of confusion with some of my colleagues.